### PR TITLE
WordpressDotCom: put downloaded assets into /year/month subfolders to reduce clashes

### LINKED ISSUE #435: WordpressDotCom importer overwrites assets
I tried importing my WordpressDotCom website into Jekyll, but I had 350 images get lost/overwritten. This was due to the authoring tool I was using naming all images:  image1, image2, image3.  
e.g.
```
/2020
  /01
    image1.jpg
    image2.jpg
    image3.jpg
  /02
    image1.jpg
    image2.jpg
    image3.jpg
```

but the WordpressDotCom importer, puts all images into /assets which results in them overwriting each other.

I see that there is a similar issue from 6 years ago that went stale #173

### PR DESCRIPTION
**Pull request by DavidBurela**
This PR resolves #435 

A small change to put downloaded assets into /year/month subfolder. This is done by using the date of the post as the subfolder in the assets folder. This keeps the assets chronologically with the post, and reduces clashes as this is the same as how Wordpress store their files.

**Before:**
When I ran the import tool on my blog, I had 350 clashes of images being overwritten.

**After:**
I successfully ran this on my blog https://DavidBurela.wordpress.com, which has 13 years, 230 posts. I only ended up with 5 clashes (which was due to me referencing images in other posts).

*Note:* I have never coded in Ruby before. I naively implemented it with the smallest number of changes that I could to reduce bugs.

### CODE REVIEW

File: lib/jekyll-import/importers/wordpressdotcom.rb
Comment (ashmaroli) [inline lib/jekyll-import/importers/wordpressdotcom.rb:25]: This changes the signature of a public method. Will you be able to pass the `published_at` attribute without changing the arity (no. of arguments a method is called with)?

File: lib/jekyll-import/importers/wordpressdotcom.rb
Comment (ashmaroli) [inline lib/jekyll-import/importers/wordpressdotcom.rb:35]: `datefolder` and `withdate` aren't exactly valid words, are they?
```suggestion
          date_folder = published_at.strftime("/%Y/%m")
          assets_with_date_folder = format("%s%s", assets_folder, date_folder)
```

File: lib/jekyll-import/importers/wordpressdotcom.rb
Comment (ashmaroli) [inline lib/jekyll-import/importers/wordpressdotcom.rb:39]: Wouldn't `File.join("{{ site.baseurl }}", dst)` return the same output as ln#38?

File: lib/jekyll-import/importers/wordpressdotcom.rb
Comment (ashmaroli) [reply lib/jekyll-import/importers/wordpressdotcom.rb]: Do ensure that all occurrences are renamed. I highlighted just a couple of them.

Comment (DavidBurela) [reply lib/jekyll-import/importers/wordpressdotcom.rb]: agree with this. It wasn't consistent with other variables

File: lib/jekyll-import/importers/wordpressdotcom.rb
Comment (DavidBurela) [reply lib/jekyll-import/importers/wordpressdotcom.rb:25]: The call to this method on line 199 has been updated with the extra parameter
Unless there is another place I'm not aware of?

File: lib/jekyll-import/importers/wordpressdotcom.rb
Comment (DavidBurela) [reply lib/jekyll-import/importers/wordpressdotcom.rb]: Your right, that is WAY simpler.  I just slotted my extra variable into the existing code concat, but rewriting it is cleaner. Updated the PR with this

**Comment by DavidBurela** [comment]
Pulled date subfolder logic earlier in execution chain. This keeps method signature intact. Guarded it with an if statement, so it should also be safer than the original.

Tested it on a fresh import of 161 posts with 837 images.

File: lib/jekyll-import/importers/wordpressdotcom.rb
Comment (DavidBurela) [reply lib/jekyll-import/importers/wordpressdotcom.rb]: Restored the method signature.  moved the logic earlier and guarded it in case there are pages or posts without a publish date

File: lib/jekyll-import/importers/wordpressdotcom.rb
Comment (ashmaroli) [inline lib/jekyll-import/importers/wordpressdotcom.rb:202]: `date_folder` isn't used elsewhere so we can avoid creating it. But then again, `assets_sub_folder` will be used only `if fetch`. So wrap the whole thing:
```suggestion
            if fetch
              # Put the images into a /yyyy/mm/ subfolder to reduce clashes
              assets_dir_path = if item.published_at
                                  File.join(assets_folder, item.published_at.strftime("/%Y/%m"))
                                else
                                  assets_folder
                                end

              download_images(item.title, content, assets_dir_path)
            end
```

**Comment by DavidBurela** [comment]
Thanks for your patience on this PR. As I mentioned, I've never used Ruby. So your assistance has been appreciated.

**Comment by ashmaroli** [comment]
> As I mentioned, I've never used Ruby.

Yes, that's the reason I walked you through this.

I did not merge this yet because I'm not sure if everyone would want their assets put into this particular subdir path.

**Comment by ashmaroli** [comment]
@jekyllbot: merge +fix

**Comment by ashmaroli** [comment]
Thank you @DavidBurela

**Review by soulfulAdam — APPROVED** [review]

**Comment by soulfulAdam** [comment]
.

**Pull request merged**
